woodwork/tests/accessor/test_indexers.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+import pytest
+
+from woodwork.indexers import _iLocIndexerAccessor
+from woodwork.logical_types import Categorical
+from woodwork.tests.testing_utils import to_pandas, xfail_dask_and_koalas
+from woodwork.utils import import_or_none
+
+dd = import_or_none('dask.dataframe')
+
+
+def test_iLocIndexer_class_error(sample_df_dask, sample_series_dask):
+    with pytest.raises(TypeError, match="iloc is not supported for Dask DataFrames"):
+        _iLocIndexerAccessor(sample_df_dask)
+
+    with pytest.raises(TypeError, match="iloc is not supported for Dask Series"):
+        _iLocIndexerAccessor(sample_series_dask)
+
+
+# def test_iLocIndexer_class(sample_df):
+#     if dd and isinstance(sample_df, dd.DataFrame):
+#         pytest.xfail('iloc is not supported with Dask inputs')
+#     dt = ww.DataTable(sample_df)
+#     ind = _iLocIndexer(dt)
+#     pd.testing.assert_frame_equal(to_pandas(ind.underlying_data), to_pandas(sample_df))
+#     pd.testing.assert_frame_equal(to_pandas(ind[1:2].to_dataframe()), to_pandas(sample_df.iloc[1:2]))
+#     assert ind[0, 0] == 0
+
+
+def test_iloc_column(sample_series):
+    xfail_dask_and_koalas(sample_series)
+    series = sample_series.astype('category')
+    logical_type = Categorical
+    semantic_tags = ['tag1', 'tag2']
+    description = 'custom column description'
+    metadata = {'meta_key': 'custom metadata'}
+    series.ww.init(logical_type=logical_type,
+                   semantic_tags=semantic_tags,
+                   description=description,
+                   metadata=metadata)
+
+    sliced = series.ww.iloc[2:]
```

Comment:
Can we check that changing the schema of `sliced` doesn't also change `series`'  attributes? The ones I'm potentially worried about are `semantic_tags` and `metadata`.

Reply:
Yeah, good call. Probably need to add some copy operations. I'll add a test for this.